clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy

```diff
@@ -203,6 +204,9 @@ class AutoScalingWorker {
     // Favor subnetIds over availability zones
     def subnetIds = subnetIds?.join(',')
     if (subnetIds) {
+      if (securityGroups && maxSecurityGroupsPerInstance && securityGroups.size() > maxSecurityGroupsPerInstance) {
+        throw new IllegalArgumentException("An instance can have at most ${maxSecurityGroupsPerInstance} security groups")
```

Comment:
Not sure if calling out `instance` in the error message would be confusing to users given that it's happening in the context of them creating a server group?